### PR TITLE
Update CAPI to v1.2.2

### PR DESCRIPTION
**Pull request by Ankitasw**
**What type of PR is this?**
/area release

**What this PR does / why we need it**:
This PR bumps CAPI from v1.2.0 to v1.2.2. This PR also adds SSA immutability E2E test for AWSMachineTemplate.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [X] adds or updates e2e tests

### CODE REVIEW

File: test/e2e/suites/unmanaged/unmanaged_CAPI_clusterclass_test.go [hunk 1]
```diff
@@ -91,4 +91,50 @@ var _ = ginkgo.Context("[unmanaged] [Cluster API Framework] [ClusterClass]", fun
...
+				// BootstrapConfigTemplate of all MachineDeploymentClasses of the ClusterClass after the initial Cluster creation.
+				// The test verifies that these fields are rolled out to the MachineDeployments.
+				ModifyMachineDeploymentBootstrapConfigTemplateFields: map[string]interface{}{
+					"spec.template.spec.verbosity": int64(4),
+				},
+
+				// ModifyMachineDeploymentInfrastructureMachineTemplateFields are the fields which will be set on the
+				// InfrastructureMachineTemplate of all MachineDeploymentClasses of the ClusterClass after the initial Cluster creation.
+				// The test verifies that these fields are rolled out to the MachineDeployments.
+				ModifyMachineDeploymentInfrastructureMachineTemplateFields: map[string]interface{}{
```
Comment (sedefsavas) [inline test/e2e/suites/unmanaged/unmanaged_CAPI_clusterclass_test.go:127]: Any change verifies the webhooks changes for the dry run as AWSMachineTemplate is immutable.
Does this look good to you @sbueringer ?

**Comment by Ankitasw** [comment]
@sedefsavas not sure why unit tests are failing here, it passes locally.

File: test/e2e/suites/unmanaged/unmanaged_CAPI_clusterclass_test.go (hunk 1 above)
Comment (sbueringer) [reply test/e2e/suites/unmanaged/unmanaged_CAPI_clusterclass_test.go:127]: Yes, looks perfect.

If you run the e2e test and look in artifacts you should see that the AWSMachineTemplate referenced in the ClusterClass at the end of the test should have those tags (I can double check if you can point me to a artifacts folder)

Comment (Ankitasw) [reply test/e2e/suites/unmanaged/unmanaged_CAPI_clusterclass_test.go:127]: Thanks @sbueringer, I will run the E2E tests and let you know.

Comment (Ankitasw) [reply test/e2e/suites/unmanaged/unmanaged_CAPI_clusterclass_test.go:127]: The tags are getting updated successfully. https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-aws/3739/pull-cluster-api-provider-aws-e2e/1572998364616724480/artifacts/clusters/bootstrap/resources/clusterclass-changes-4t45xq/AWSMachineTemplate/clusterclass-changes-3sac4j-md-0-infra-qnlw9.yaml

**Comment by Ankitasw** [comment]
Some issue is ocurring with machine pool tests, I will take a look.

File: test/e2e/suites/unmanaged/unmanaged_CAPI_clusterclass_test.go (hunk 1 above)
Comment (sbueringer) [reply test/e2e/suites/unmanaged/unmanaged_CAPI_clusterclass_test.go:127]: Perfect!

**Comment by Ankitasw** [comment]
/hold
waiting for #3743 to merge

**Comment by Ankitasw** [comment]
GPU tests passing locally but failing in the PR.

**Comment by sedefsavas** [comment]
/close
in favor of https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/3747

**Pull request closed without merging**